cmd/mt-update-ttl/main.go

```diff
@@ -0,0 +1,167 @@
+	cassandraHostSelectionPolicy = flag.String("cassandra-host-selection-policy", "roundrobin", "")
+	cassandraTimeout             = flag.Int("cassandra-timeout", 1000, "cassandra timeout in milliseconds")
+	cassandraConcurrency         = flag.Int("cassandra-concurrency", 20, "max number of concurrent reads to cassandra.")
+	cassandraRetries             = flag.Int("cassandra-retries", 0, "how many times to retry a query before failing it")
+	cqlProtocolVersion           = flag.Int("cql-protocol-version", 4, "cql protocol version to use")
+
+	cassandraSSL              = flag.Bool("cassandra-ssl", false, "enable SSL connection to cassandra")
+	cassandraCaPath           = flag.String("cassandra-ca-path", "/etc/raintank/ca.pem", "cassandra CA certificate path when using SSL")
+	cassandraHostVerification = flag.Bool("cassandra-host-verification", true, "host (hostname and server cert) verification when using SSL")
+
+	cassandraAuth     = flag.Bool("cassandra-auth", false, "enable cassandra authentication")
+	cassandraUsername = flag.String("cassandra-username", "cassandra", "username for authentication")
+	cassandraPassword = flag.String("cassandra-password", "cassandra", "password for authentication")
+
+	verbose = flag.Bool("verbose", false, "show every record being processed")
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "mt-update-ttl [flags] ttl table-in [table-out]")
+		fmt.Fprintln(os.Stderr)
+		fmt.Fprintln(os.Stderr, "Adjusts the data in Cassandra to use a new TTL value. The TTL is applied counting from the timestamp of the data")
+		fmt.Fprintln(os.Stderr, "If table-out not specified or same as table-in, will update in place. Otherwise will not touch input table and store results in table-out")
+		fmt.Fprintln(os.Stderr, "In that case, it is up to you to assure table-out exists before running this tool")
+		fmt.Fprintln(os.Stderr, "Not supported yet: for the per-ttl tables as of 0.7, automatically putting data in the right table")
+		fmt.Println("Flags:")
+		flag.PrintDefaults()
+		os.Exit(-1)
+	}
+	flag.Parse()
+
+	if flag.NArg() < 2 {
+		flag.Usage()
+	}
+
+	ttl := int(dur.MustParseUNsec("ttl", flag.Arg(0)))
+	tableIn, tableOut := flag.Arg(1), flag.Arg(1)
+	if flag.NArg() == 3 {
+		tableOut = flag.Arg(2)
+	}
+
+	session, err := NewCassandraStore(*cassandraAddrs, *cassandraKeyspace, *cassandraConsistency, *cassandraCaPath, *cassandraUsername, *cassandraPassword, *cassandraHostSelectionPolicy, *cassandraTimeout, *cassandraConcurrency, *cassandraRetries, *cqlProtocolVersion, *cassandraSSL, *cassandraAuth, *cassandraHostVerification)
+
+	if err != nil {
+		panic(fmt.Sprintf("Failed to instantiate cassandra: %s", err))
+	}
+
+	update(session, ttl, tableIn, tableOut)
+}
+
+func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, concurrency, retries, protoVer int, ssl, auth, hostVerification bool) (*gocql.Session, error) {
```

Comment:
this function started as a modified variant of `NewCassandraStore` in `mdata`, but you're right, in this case it's nicer to just use the global vars. fixed.